bodhi/tests/server/consumers/test_masher.py

```diff
@@ -1588,3 +1749,182 @@ def test_urlerror(self, urlopen, sleep, publish):
+        config.get.return_value = "/tmp/bodhi/pungi"
+        pungi_conf_path = self.wrapper._get_pungi_conf_path()
+        assert pungi_conf_path == "/tmp/bodhi/pungi/fedora-modular-example.conf"
+
+    def test_get_compose_dir(self):
+        compose_dir = self.wrapper._get_compose_dir("/tmp")
+        assert compose_dir == "/tmp/compose/Server"
+
+    @mock.patch("bodhi.server.consumers.masher.PungiMasherThread._get_compose_dir")
+    @mock.patch("bodhi.server.consumers.masher.sanity_check_repodata")
+    @mock.patch("os.listdir")
+    def test_sanity_check_repo(self, list_dir, sanity_check_repodata, get_compose_dir):
+        list_dir.return_value = ["x86_64"]
+        result = self.wrapper.sanity_check_repo()
+        sanity_check_repodata.assert_called_once()
+        get_compose_dir.assert_called_once()
+        assert result
+
+    @mock.patch("bodhi.server.consumers.masher.PungiMasherThread._get_compose_dir")
+    @mock.patch("bodhi.server.consumers.masher.sanity_check_repodata")
+    @mock.patch("os.listdir")
+    @mock.patch("os.path.islink")
+    def test_sanity_check_repo_symlink_exception(self, islink, list_dir,
+                                                 sanity_check_repodata, get_compose_dir):
+        list_dir.return_value = ["package.rpm"]
+        islink.return_value = True
+        with pytest.raises(Exception):
+            self.wrapper.sanity_check_repo()
+            sanity_check_repodata.assert_called_once()
+            get_compose_dir.assert_called_once()
+            self.log.error.assert_called_once()
+
+    @mock.patch("bodhi.server.consumers.masher.PungiMasherThread._get_compose_dir")
+    @mock.patch("bodhi.server.consumers.masher.sanity_check_repodata")
+    @mock.patch("os.listdir")
+    def test_sanity_check_repo_repodata_exception(self, list_dir, sanity_check_repodata,
+                                                  get_compose_dir):
+        list_dir.return_value = ["x86_64"]
+        sanity_check_repodata.side_effect = Exception("Repodata validation failure!")
+        with pytest.raises(Exception):
+            self.wrapper.sanity_check_repo()
+            sanity_check_repodata.assert_called_once()
+            get_compose_dir.assert_called_once()
+            self.log.error.assert_called_once()
+
+    @mock.patch("bodhi.server.consumers.masher.PungiMasherThread._get_compose_dir")
+    @mock.patch("bodhi.server.consumers.masher.PungiMetadata")
+    def test_generate_update_info(self, pungi_metadata, get_compose_dir):
+        self.wrapper.release = mock.Mock()
+        self.wrapper.release.return_value = "f27-modular-updates"
+        self.wrapper.generate_updateinfo()
+
+        pungi_metadata.assert_called_once()
+        get_compose_dir.assert_called_once()
+
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_skip_mash(self, *args):
+        masher_mock = mock.create_autospec(self.wrapper)
+        masher_mock.work = lambda: PungiMasherThread.work(masher_mock)
```

Comment:
I'm surprised that ```flake8``` isn't upset at this - it usually gets upset if lambdas are assigned to variables, because it's contrary to the idea of lambdas (they are supposed to be anonymous, and if you assign them to a varaible they are no longer anonymous.)

Reply:
Heh.  Yeah.  :p 

`flake8` is boss.